kafka-client/src/main/java/dev/responsive/kafka/internal/db/mongo/MongoKVTable.java

```diff
@@ -150,21 +164,63 @@ public MongoKVFlushManager init(final int kafkaPartition) {
   }
 
   @Override
-  public byte[] get(final int kafkaPartition, final Bytes key, final long minValidTs) {
-    final KVDoc v = docs.find(Filters.and(
-        Filters.eq(KVDoc.ID, keyCodec.encode(key)),
-        Filters.gte(KVDoc.TIMESTAMP, minValidTs)
-    )).first();
-    return v == null ? null : v.getValue();
+  public byte[] get(final int kafkaPartition, final Bytes key, final long streamTimeMs) {
+
+    // Need to post-filter if value is needed to compute ttl
```

Comment:
I actually think with the approach suggested above, we can do it all server side because `expire_ts` is already a dedicated column so we don't need to recompute it. We could actually consider doing the same thing in Scylla by inserting an extra column for expire_ts (they just wouldn't create an index on it).